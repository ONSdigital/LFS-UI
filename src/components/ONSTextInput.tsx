import React, {ChangeEvent, Component} from 'react';

interface Props {
    label?: string,
    id?: string,
    password?: boolean,
    onChange?: (e: ChangeEvent<HTMLInputElement>, ...args: any[]) => void,
    placeholder?: string,
    fit?: boolean,
    autoFocus?: boolean,
}

interface State {
    value: string
}

export class ONSTextInput extends Component <Props, {}> {
    value: string = "";

    constructor(props: Props) {
        super(props);
        this.state = {value: ""};
    }

    handleChange = (e: ChangeEvent<HTMLInputElement>) => {
        if (this.props.onChange !== undefined) {
            this.props.onChange(e, this.props.label);
        }
        this.value = e.target.value;

    };

    render() {
        return (
            <p className="field">
                {this.props.label !== undefined && <label className="label">{this.props.label}</label>}
                <input style={{width: this.props.fit === true ? "unset" : ""}}
                       autoFocus={this.props.autoFocus === true}
                       type={this.props.password === true ? "password" : "text"} id={this.props.id}
                       className="input input--text input-type__input " placeholder={this.props.placeholder}
                       onChange={(e) => this.handleChange(e)}/>
            </p>
        );
    }
}
