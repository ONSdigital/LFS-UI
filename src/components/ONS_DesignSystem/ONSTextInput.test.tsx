import React from 'react';
import { shallow, mount } from 'enzyme';
import {ONSTextInput} from "./ONSTextInput";
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

describe("ONS Text Input Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const Props = {
    }

    function wrapper(render: any, props: any) {
        return render(
            <ONSTextInput
                label={props.label}
                id={props.id}
                password={props.password}
                onChange={props.onChange}
                placeholder={props.placeholder}
                fit={props.fit}
                autoFocus={props.autoFocus}
                value={props.value}
                autoComplete={props.autoComplete}
                onClick={props.onClick}>
            </ONSTextInput>
            
        )
    }

    it("should render correctly", () => expect(wrapper(shallow, Props).exists()).toEqual(true));



})