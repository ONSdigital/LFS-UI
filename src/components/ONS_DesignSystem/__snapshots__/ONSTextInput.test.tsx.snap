// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Text Input Test matches Snapshot 1`] = `ShallowWrapper {}`;
